<?xml version="1.0"?>
<launch>

    <!-- get the arguments from  -->
    <arg name="items" default="items.yaml" doc="Item Dictionary" />
    <arg name="debug" default="False" doc="Whether debug is on" />
    <arg name="suffix" default="0" doc="Suffix to add to nodes" />
    <arg name="tag_topic" default="/tag_detections" doc="April tag topic" />
    <arg name="arrange_topic" default="/desired_arrange" doc="Arragement setpoint topic" />
    <arg name="rate" default="30" doc="Rate (in Hz) to run the controller (this should roughly match your camera's framerate)" />
    <arg name="pctrl" default="False" doc="Using pressure controler or not"/>
    <arg name="arm" default="False" doc="Using arm or not"/>    
    <arg name="speed_factor" default="1.0" doc="the speed to run the trajectory at" />
    <arg name="reps" default="1" doc="The number of reps" />
    <arg name="arm_traj" default="False" doc="Arm trajectory file"/>
    <arg name="traj_profile" default="all_traj.yaml" doc="Finger trajectory file" />
    <arg name="traj" default="start_grasp" doc="Finger trajectory file" />


    <!-- Generate helper arguments -->
    <arg name="_suffix_helper"     if="$(arg suffix)" value="_$(arg suffix)"  />
    <arg name="_suffix_helper" unless="$(arg suffix)" value=""  />

    <!-- Start the main serial sender -->
    <node name="tag2arrange_node$(arg _suffix_helper)" pkg="tag2arrange" type="controller_node.py" respawn="false"
        output="screen">
        <rosparam file="$(find tag2arrange)/config/items/$(arg items)" command="load"/>
        <param name="DEBUG" type="bool" value="$(arg debug)"/>
        <param name="tag_topic" type="str" value="$(arg tag_topic)"/>
        <param name="arrange_topic" type="str" value="$(arg arrange_topic)"/>
        <param name="controller_rate" type="double" value="$(arg rate)"/>
        <param name="pctrl" type="bool" value="$(arg pctrl)"/>
        <param name="arm" type="bool" value="$(arg arm)"/>
        <rosparam file="$(find tag2arrange)/config/trajectories/$(arg traj_profile)" command="load"/>     

        
        <param name="arm_traj_profile" type="str" value="$(find tag2arrange)/config/trajectories/arm_traj/$(arg arm_traj)"/>
        <param name="speed_factor" type="double" value="$(arg speed_factor)" />
        <param name="num_reps" type="int" value="$(arg reps)" />
    </node>

</launch>